lib/cn/route.c

```diff
@@ -86,25 +88,32 @@ route_map_find(struct route_map *map, const void *key, uint keylen)
             if (dir >= 0 && *link)
                 node = *link;
         } else {
-            return this;
+            return gt ? rb_entry(rb_next(*link), struct route_node, rtn_node) : this;
         }
     }
 
+    node = gt ? rb_next(node) : node;
     return rb_entry(node, struct route_node, rtn_node);
 }
 
 struct route_node *
 route_map_lookup(struct route_map *map, const void *pfx, uint pfxlen)
 {
-    return map ? route_map_find(map, pfx, pfxlen) : NULL;
+    return map ? route_map_find(map, pfx, pfxlen, false) : NULL;
+}
+
+struct route_node *
+route_map_lookupGT(struct route_map *map, const void *pfx, uint pfxlen)
```

Comment:
How about naming it route_map_lookup_next()?

Reply:
Seems like you could also do a route_map_lookup() followed by a route_node_next()? It's fine if this is called in the perf. path and you would like to avoid a second call into routemap.

Reply:
Yes, I've added a comment to each of the two lookup variants describing this detail.